lib/internal/test_runner/tap_stream.js

```diff
@@ -115,7 +123,23 @@ class TapStream extends Readable {
 
     line += '\n';
 
-    this.#tryPush(line);
+    this.#tryPushString(line);
+  }
+
+  #emit(type, data) {
+    this.emit(type, data);
+    this.#tryPushObject({ type, data });
+  }
+
+  #tryPushString(str) {
+    if (!this._readableState.objectMode) {
+      this.#tryPush(str);
+    }
+  }
+  #tryPushObject(obj) {
+    if (this._readableState.objectMode) {
+      this.#tryPush(obj);
+    }
```

Comment:
I agree (and raised that [here](https://github.com/nodejs/node/pull/45712#discussion_r1038116033)).
regarding building on top of the TAP parser - that will limit us to only support reporters with `--test` since running a test file doesn't involve parsing TAP

Reply:
We *could* use the TAP parser without `--test`... or maybe we should move away from TAP everything and just make that the default reporter. I think it would be mostly internal changes that would not be noticeable by end users.

Reply:
why serialize TAP just so we are able to de-serialize it?